pkg/services/ngalert/api/hooks.go

```diff
@@ -0,0 +1,38 @@
+package api
+
+import (
+	"github.com/grafana/grafana/pkg/api/response"
+	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
+)
+
+type RequestHandlerFunc func(*contextmodel.ReqContext) response.Response
+
+type Hooks struct {
+	hooks map[string]RequestHandlerFunc
+}
+
+// NewHooks creates an empty set of request handler hooks. Hooks can be used
+// to replace handlers for specific paths.
+func NewHooks() *Hooks {
+	return &Hooks{
+		hooks: make(map[string]RequestHandlerFunc),
+	}
+}
+
+// Add creates a new request hook for a path, causing requests to the path to
+// be handled by the hook function, and not the original handler.
+func (h *Hooks) Set(path string, hook RequestHandlerFunc) {
```

Comment:
```suggestion
func (h *Hooks) Set(path string, hook RequestHandlerFunc) {
  h.logger.Info("setting hook override for the specified route", "path", path)
```

Reply:
Similar concept - I'd like to verify this when grafana start to make sure my hooks are set correctly.